### ISSUE TITLE
Configuration Issues

### ISSUE THREAD

**Issue by sidferreira** [wallabyjs/quokka#81]
### Issue description or question
My Quokka is totally unstable:

- triggers while I'm typing and gets stuck in errors (like when I'm removing a `//` it triggers when it has only one slash and I have to restart it)
- Conflicts with my `.babelrc` or `package.json's babel` configuration
- Give unexpected token our of nowhere

Package.json:
```
{
  "name": "...",
  "version": "0.0.1",
  "description": "Cloud Functions for Firebase",
  "private": true,
  "scripts": {
    "prepare": "babel ./*.es7 --presets latest --retain-lines -d ./"
  },
  "dependencies": {
    "axios": "^0.16.2",
    "babel-runtime": "^6.26.0",
    "cookie-parser": "^1.4.3",
    "cors": "^2.8.4",
    "crypto": "0.0.3",
    "firebase-admin": "~4.2.1",
    "firebase-functions": "^0.5.7",
    "firebase-util": "^0.2.5",
    "moment": "^2.18.1",
    "simple-oauth2": "^1.3.0"
  },
  "devDependencies": {
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-preset-latest": "^6.24.1",
    "standard": "^10.0.3"
  },
  "quokka": {
    "babel": {
      "env": "dev"
    }
  }
}
```

.babelrc (disabled for now)
````
{
  "presets": [
    "es2017"
  ],
  "plugins": [
    [
      "transform-runtime",
      {
        "polyfill": false,
        "regenerator": true
      }
    ]
  ]
}
```

Latest token error: 
```
​​Unexpected token (​​
  at ​​​createScript​​​ ​vm.js:74​
  at ​​​Object.runInThisContext​​​ ​vm.js:116​
```

Code being tested:
```
if (__DEV__) {
  const onUpdateCompanyUserTest = async () => {
    try {
      const companyId = '...'
      const email = '...'
      const path = `company/${companyId}/users/${email}`
      const ref = await admin.database().ref(path) 
      const userSnapshot = await ref.once('value') /*?*/
      const userdata = userSnapshot.val() /*?*/
      const data = new functions.database.DeltaSnapshot(null, null, userdata, userdata, path) /*?*/
      const params = { companyId, email } /*?*/
      // await exports.onUpdateCompanyUser({ data, params }) /*?*/
    } catch(e) {
      console.log(e) /*?*/
      return false
    }
    return true
  }
  onUpdateCompanyUserTest() /*?*/
}
```
Notes:

- `const ref` isn't a promise, but seem that will work only with it
- `const userSnapshot` gets the result some times, outputs it, but the function keeps inside a loop

Bottom line: Loved it, want to use it, but for now I'll drop it

**Comment by ArtemGovorov** [comment]
So I went ahead and created this sample repo with your configs and code: https://github.com/ArtemGovorov/quokka-81

Here is what I have discovered:

> Conflicts with my .babelrc or package.json's babel configuration

This is because your `.babelrc` is not really correct. With your `.babelrc`, generated code contains ES6 `import`, and as you may know node.js doesn't support ES6 imports.

Now you also have the `prepare` script in your `package.json` that is NOT a babel configuration that any tool respects (except `npm` when you run it). In the command, you are using the `latest` preset (that is correct for node), so the script generates valid node compatible JavaScript.

You either need to correct your .babelrc to do the same, or just override it in your Quokka config: like I did in the sample repo: https://github.com/ArtemGovorov/quokka-81/blob/master/package.json#L32.

> triggers while I'm typing and gets stuck in errors (like when I'm removing a // it triggers when it has only one slash and I have to restart it)
> Give unexpected token our of nowhere

You may try the sample repo and you will not et any of these issues. Quokka does trigger when you remove on slash (as designed, it may run on each key stroke depending on how fast you type), and displays the error (if there's an error), but when you remove the other one, it is supposed to run again and clear things up.

![q](https://user-images.githubusercontent.com/979966/29651055-fad7ba1a-88e1-11e7-889c-2d9bb17f9969.gif)

Because the code in the repo is exactly yours (except mocking dependencies), and it's working (you may try it), my suspicion is that some of your real project dependencies, like firebase may be caching something on a process level, so that re-runs fail. Another issue maybe that your DB hangs, so after you fix a syntax error in Quokka (like remove the single slash that is left), your code takes too long to execute (because for example because the DB is hanging) so it may look like Quokka gets stuck in errors

To avoid the caching, we have added a new flag (called `recycle`) in the latest Quokka v 1.0.54. The flag makes quokka to recreate node processes, instead of reusing one. It's a bit slower than reusing the process, but may be the only option for scenario with DBs like yours.  I have [added the flag to the config](https://github.com/ArtemGovorov/quokka-81/blob/master/package.json#L29) in the sample repo. Try running your code with the flag set to `true` and let know how it goes.

**Comment by sidferreira** [comment]
@artemgororov shame on me. Long week, computer broke and rhe greatest tool ever wasn't working as expected. My bad. Will follow your specs and feedback in 2 hours

**Comment by sidferreira** [comment]
@ArtemGovorov I'm using a freaking bluetooth keyboard (low quality one that I'm about to replace) and the typing speed isn't regular. Maybe it causes some of the issues too.

About the hanging, it still happens. Can you explain me how to debug it? So I can spend my time on it instead of yours?

**Comment by ArtemGovorov** [comment]
> About the hanging, it still happens. 

Does it happen in the sample repo I have created?
Maybe it's the database being not very responsive (due to the frequent calls)?

**Comment by sidferreira** [comment]
@ArtemGovorov seem that firebase is the thing. Maybe a process is hanged and that's why Quokka keeps running (that [--=--] animation) and never finishes mapping.
